index.js

```diff
@@ -0,0 +1,46 @@
+const PermissionTree = require('./src/PermissionTree');
+
+module.exports = function(app, optionsToMerge) {
+    const options = Object.assign({ mountPath: '/getUserPermissionTree' }, optionsToMerge);
+    const permissionTree = new PermissionTree(app.models, app.remotes());
+
+    app.get(options.mountPath, async(req, res) => {
+        try {
+            const user = await new Promise((resolve, reject) => {
+                app.models.AccessToken.findForRequest(req, {}, (error, accesstoken) => {
+                    if (error) reject(error);
```

Comment:
In case of an error the promise is rejected AND resolved... (it would even try to send multiple responses, one by the middleware and one by the error handler!).